lib/tar.ml

```diff
@@ -196,10 +196,8 @@ module Header = struct
     Cstruct.blit_from_string v 0 buf hdr_magic_off sizeof_hdr_magic
 
   let _get_hdr_version buf =
-    unmarshal_int
-      (Cstruct.to_string ~off:hdr_version_off ~len:sizeof_hdr_version buf)
+    Cstruct.to_string ~off:hdr_version_off ~len:sizeof_hdr_version buf
   let set_hdr_version buf v =
-    let v = marshal_int v sizeof_hdr_version in
     Cstruct.blit_from_string v 0 buf hdr_version_off sizeof_hdr_version
```

Comment:
hmm, this makes the `hdr_version` pretty "unsafe", all other accessors directly embed the marshal_string/int / unmarshal_string/int calls (that pad the string with trailing zeroes / strip the trailing zeroes).

I'm not entirely sure whether that fill with zeroes is needed, since cstruct 2.2.0 (released 2016-06-30) started to zero out a Cstruct.create - and I suspect this code predates that behaviour.

Reply:
But padding may be good to have, otherwise the `blit_from_string` if the passed `v` exceeds `sizeof_hdr_version` may lead to trouble.